week1/reposts/boccato/Main.hs

```diff
@@ -0,0 +1,38 @@
+module Main where
+
+import Control.Monad
+import Data.List
+import System.Random
+import Text.Printf
+
+nWinners = 7
+nSimulations = 10000
+nCurrentReposts = 1000000
+
+takeDistinct :: Eq a => Int -> [a] -> [a]
```

Comment:
I see. But since we have to check whether `x` is in the list we are accumulating about `n` times or more, shouldn't that procedure also take about n^2 too? Or n log n?